src/QDL/usblib.js

```diff
@@ -0,0 +1,186 @@
+import { concatUint8Array, sleep } from "./utils";
+
+const vendorID = 0x05c6;
+const productID = 0x9008;
+const QDL_USB_CLASS = 0xff;
+
+const BULK_TRANSFER_SIZE = 16384;
+
+export class UsbError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = "UsbError";
+  }
+}
+
+export class usbClass {
+  device;
+  epIn;
+  epOut;
+
+  _registeredUsbListeners;
+
+  constructor() {
+    this.device = null;
+    this.epIn = null;
+    this.epOut = null;
+    this._registeredUsbListeners = null;
+    this.maxSize = 512;
+  }
+
+
+  get connected() {
+    return (
+      this.device !== null &&
+      this.device.opened &&
+      this.device.configurations[0].interfaces[0].claimed
+    );
+  }
+
+
+  async _validateAndConnectDevice() {
+    let ife = this.device?.configurations[0].interfaces[0].alternates[0];
+    if (ife.endpoints.length !== 2) {
+      throw new UsbError("Attempted to connect to null device");
+    }
+
+    this.epIn = null;
+    this.epOut = null;
+
+    for (let endpoint of ife.endpoints) {
+      if (endpoint.type !== "bulk") {
+        throw new UsbError("Interface endpoint is not bulk");
+      }
+      if (endpoint.direction === "in") {
+        if (this.epIn === null) {
+          this.epIn = endpoint;
+        } else {
+          throw new UsbError("Interface has multiple IN endpoints");
+        }
+      } else if (endpoint.direction === "out") {
+        if (this.epOut === null) {
+          this.epOut = endpoint;
+        } else {
+          throw new UsbError("Interface has multiple OUT endpoints");
+        }
+      }
+      this.maxSize = this.epIn.packetSize;
+    }
+    console.log("Endpoints: in =", this.epIn, ", out =", this.epOut);
+
+    try {
+        await this.device?.open();
+        await this.device?.selectConfiguration(1);
+        try {
+          await this.device?.claimInterface(0);
+        } catch(error) {
+          await this.device?.reset();
+          await this.device?.forget();
+          await this.device?.close();
+          throw error;
+        }
+    } catch (error) {
+      throw error;
+    }
+  }
+
+
+  async connect() {
+    console.log("Trying to connect Qualcomm device")
+    let devices = await navigator.usb.getDevices();
+    console.log("Found these USB devices:", devices);
+
+    this.device = await navigator.usb.requestDevice({
+      filters: [
+        {
+          vendorID  : vendorID,
+          productID : productID,
+          classCode : QDL_USB_CLASS,
+        },
+      ],
+    });
+    console.log("USing USB device:", this.device);
+
+    if (!this._registeredUsbListeners){
```

Comment:
keep the style consistent
```suggestion
    if (!this._registeredUsbListeners) {
```